### PR TITLE
Increase compiler's strictness

### PR DESCRIPTION
**Pull request by emilianoLeite**
This improves type-safety and the developer experience

### CODE REVIEW

File: src/exporters/insomnia/types/environment.ts [hunk 1]
```diff
@@ -1,3 +0,0 @@
-class Environment {
-    
-}
```
Comment (emilianoLeite) [inline src/exporters/insomnia/types/environment.ts:3]: Was not being used

File: src/tests/sample.test.ts [hunk 2]
```diff
@@ -1,6 +0,0 @@
-describe('calculate', function() {
-    it('add', function () {
-        let result = 5+2;
-        expect(result).toBe(7);
-    });
-});
```
Comment (emilianoLeite) [inline src/tests/sample.test.ts:6]: Typescript was complaining since this file "wasn't a module"

File: tsconfig.json [hunk 3]
```diff
@@ -1,8 +1,15 @@
...
   "compilerOptions": {
-    "target": "es3",
+    "forceConsistentCasingInFileNames": true,
+    "incremental": true,
+    "isolatedModules": true,
     "moduleResolution": "node",
-    "outDir": "./dist"
+    "noImplicitReturns": true,
+    "noFallthroughCasesInSwitch": true,
+    "noImplicitAny": false,
```
Comment (emilianoLeite) [inline tsconfig.json:9]: Ideally we should remove this line, but it is **a lot** of work to type _everything_ 😅 
```suggestion
```

File: src/exporters/insomnia/types/request.ts [hunk 4]
```diff
@@ -3,14 +3,14 @@ import RequestBody from './request_body';
...
-    parentId: string;
     created: number;
-    modified: number;
     name: string;
-    method: string = "POST";
-    url: string;
-    body: RequestBody;
+    method: HTTPMethod = "POST";
     headers: any[] = [];
+    parentId: Maybe<string>;
```
Comment (oshosanya) [inline src/exporters/insomnia/types/request.ts:10]: What's the implication of using `Maybe<string>` as against `string`?

Comment (emilianoLeite) [reply src/exporters/insomnia/types/request.ts:10]: In this case is a hand-made type (I created it [here](https://github.com/oshosanya/graphql-export/pull/10/files#diff-af565283322a1965b4ea9bfb48fe1f578631a934aadbfc7a413fe0d5e84746fbR1)). But the _concept_ comes from functional programming:
>In programming languages (more so functional programming languages) and type theory, an option type or maybe type is a polymorphic type that represents encapsulation of an optional value; e.g., it is used as the return type of functions which may or may not return a meaningful value when they are applied. It consists of a constructor which either is empty (often named `None` or `Nothing`), or which encapsulates the original data type `A` (often written `Just A` or `Some A`).

[Wikipedia reference](https://en.wikipedia.org/wiki/Option_type)

----

The way that I've written it, it means: "the value is either a `string` (in this case) or  'empty value' (represented as `null | undefined`)"

File: src/@types/global.d.ts [hunk 5]
```diff
@@ -0,0 +1,2 @@
+declare type Maybe<T> = T | null | undefined;
+declare type HTTPMethod = 'GET' | 'POST' | 'PUT' | 'DELETE'
```
Comment (emilianoLeite) [inline src/@types/global.d.ts:2]: This types are "global". They can be referenced without the need to manually import them beforehand

File: src/exporters/insomnia/types/request.ts (hunk 4 above)
Comment (emilianoLeite) [reply src/exporters/insomnia/types/request.ts:10]: I chose using this Maybe as the "path of least resistance". In reality, I think some of these fields could be completly removed (i.e. they never have values assigned)

File: tsconfig.json (hunk 3 above)
Comment (emilianoLeite) [reply tsconfig.json:9]: Update: this line is removed in #11  :tada:

File: src/exporters/insomnia/types/request.ts (hunk 4 above)
Comment (oshosanya) [reply src/exporters/insomnia/types/request.ts:10]: I get this now, makes a lot of sense

**Pull request merged**